superset-frontend/src/hooks/useDebounceValue.ts

```diff
@@ -0,0 +1,36 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+import { useState, useEffect } from 'react';
+
+export function useDebounceValue(value: string, delay = 500) {
```

Comment:
There's a `FAST_DEBOUNCE` in `src/constants.ts`.

Reply:
Why do we need this hook? Can't we just use `debounce` from `lodash`? That's what we are using elsewhere.

Reply:
yes but need a extra useMemo wrapper. this is a convenient(shared) way in order to reduce the burden.

i.e.
```
 const [value, setValue] = useState('');
 const setValueDebounced = useMemo(() => debounce(setValue, DELAY));
 ...
 setValueDebounced('updated');
```

vs 

```
 const [value, setValue] = useState('');
 const debouncedValue = useDebounceValue(value);
 ...
 setValue('updated');
```

Reply:
Can you use `debounce` inside the hook?

Reply:
`useDebounceValue` mechanism is different. 
(`debounce` makes setter in debounce state whereas useDebounceValue look received the actual value and return the debounced update value) the actual implementation of useDebounceValue hook is how original debounce is implemented. (but it's less size and easier way to handle the cancel for unmounted state)
I'm feeling using lodash/debounce in useDebounceValue hook is overkill 

(it can be like)
```
function useDebounceValue(value: string, delay = FAST_DEBOUNCE) {
  const [debouncedValue, setValue] = useState(value);
  const setDebouncedValue = useMemo(() => debounce(setValue, delay), [delay]);

  useEffect(() => {
    setDebouncedValue(value);
    return () => {
      setDebouncedValue.cancel();
    };
  }, [value, setDebouncedValue]);

  return debouncedValue;
}

```